src/snowflake/cli/api/project/util.py

```diff
@@ -143,3 +143,8 @@ def validate_version(version: str):
         raise ValueError(
             f"Project definition version {version} is not supported by this version of Snowflake CLI. Supported versions: {SUPPORTED_VERSIONS}"
         )
+
+
+def escape_like_pattern(pattern: str) -> str:
+    pattern = pattern.replace("%", r"\\}%").replace("_", r"\\_")
```

Comment:
I am curious why do we need this escape? Is that expected to replace all "%" to r"\\}%"? Do we have test case about this?

Reply:
Good point, the escape is supposed to be "%" -> "\\%" without the "}". I've added tests so that I don't miss these typos in the future 😅. In general, the escape is needed because "%" and "_" are wildcard characters in SHOW LIKE, meaning that something like "wildcard_name" would match "wildcardAname".